### ISSUE TITLE
problem of baseline

### ISSUE THREAD

**Issue by qazwsx74269** [ConvLab/ConvLab#33]
when I run this command `python run.py baseline.json onenet_rule_dqn_template train`, it displays such information in the dataframe.

![20190717-193558(eSpace)](https://user-images.githubusercontent.com/14293897/61372579-20e13100-a8ca-11e9-9845-eea2104ee385.png)

why is the total_reward always -59? I nearly didn't modify your spec. could you tell me how to adjust it to run this baseline normally?
The spec is following:
"onenet_rule_dqn_template": {
    "agent": [{
      "name": "DialogAgent",
      "nlu": {
        "name": "OneNetLU",
        "model_file": "https://convlab.blob.core.windows.net/models/onenet.tar.gz"
      },
      "dst": {
        "name": "RuleDST"
      },
      "nlg": {
        "name": "MultiwozTemplateNLG",
        "is_user": false
      },
      "state_encoder": {
        "name": "MultiWozStateEncoder"
      },
      "action_decoder": {
        "name": "MultiWozVocabActionDecoder"
      },
      "algorithm": {
        "name": "DQN",
        "action_pdtype": "Argmax",
        "action_policy": "epsilon_greedy",
        "explore_var_spec": {
          "name": "linear_decay",
          "start_val": 0.1,
          "end_val": 0.01,
          "start_step": 1000,
          "end_step": 500000
        },
        "gamma": 0.9,
        "training_batch_iter": 1000,
        "training_iter": 1,
        "training_frequency": 100,
        "training_start_step": 32,
        "normalize_state": false 
      },
      "memory": {
        "name": "Replay",
        "batch_size": 16,
        "max_size": 50000,
        "use_cer": false 
      },
      "net": {
        "type": "MLPNet",
        "hid_layers": [100],
        "hid_layers_activation": "relu",
        "clip_grad_val": null,
        "loss_spec": {
          "name": "MSELoss"
        },
        "optim_spec": {
          "name": "Adam",
          "lr": 0.00001
        },
        "lr_scheduler_spec": {
          "name": "StepLR",
          "step_size": 1000,
          "gamma": 0.999,
        },
        "update_type": "replace",
        "update_frequency": 300,
        "polyak_coef": 0,
        "gpu": true
      }
    }],
    "env": [{
      "name": "multiwoz",
      "action_dim": 300,
      "observation_dim": 392,
      "max_t": 40,
      "max_frame": 500000,
      "nlu": {
        "name": "MILU",
        "model_file": "https://convlab.blob.core.windows.net/models/milu.tar.gz"
      },
      "user_policy": {
        "name": "UserPolicyAgendaMultiWoz"
      },
      "sys_policy": {
        "name": "RuleBasedMultiwozBot"
      },
      "nlg": {
        "name": "MultiwozTemplateNLG",
        "is_user": true
      },
      "evaluator": {
        "name": "MultiWozEvaluator"
      }
    }],
    "meta": {
      "distributed": false,
      "num_eval": 100,
      "eval_frequency": 1000,
      "max_tick_unit": "total_t",
      "max_trial": 1,
      "max_session": 5,
      "resources": {
        "num_cpus": 0,
        "num_gpus": 1
      }
    }
  },

**Comment by pengbaolin** [comment]
Hi @qazwsx74269 . Because of the large action space, it is not easy for the agent to explore successful dialogue path, especially in the early training phase. 

We suggest you using *wdqn* which is DQN with warmup policy from a rule agent. E.g. rule_wdqn in demo.json

**Comment by qazwsx74269** [comment]
> Hi @qazwsx74269 . Because of the large action space, it is not easy for the agent to explore successful dialogue path, especially in the early training phase.
> 
> We suggest you using _wdqn_ which is DQN with warmup policy from a rule agent. E.g. rule_wdqn in demo.json

But it seems that the total_reward doesn't change when I run wdqn.
![20190718-143151(eSpace)](https://user-images.githubusercontent.com/14293897/61434550-d5ca2b00-a968-11e9-8e8e-c7ffa57e9373.png)

**Comment by pengbaolin** [comment]
Please try training_batch_iter=1 and training_iter = 500. 

Below is the result using above parameters:
Trial 0 session 0 milu_rule_wdqn_template_t0_s0 [eval_df] epi: 3400  t: 21  wall_t: 2842  opt_step: 2.1375e+06  frame: 50000  fps: 17.5932  total_reward: -59  avg_return: -11.11  avg_len: 15.11 avg_success: 0.35  loss: 417.891  lr: 0.000952158  explore_var: 0.0911623  entropy_coef: nan  entropy: nan  grad_norm: nan

We will update the config file soon.

**Comment by sherlock1987** [comment]
Hey, are you guys using GPU or CPU? I am a newer in Convlab, I find code only runs on CPU not GPU.

**Comment by jincli** [comment]
@sherlock1987  ConvLab only supports CPU for now.

**Comment by sherlock1987** [comment]
Okay, thanks bro!

**Comment by sherlock1987** [comment]
About wrong results from rule_wdqn. Hey guys, I use the same hyper paramaters, but my avg_success is always fluctuate, and even comes to 0  when frame is 50000, any cluse about why I falied?

Here is the results, I set the num_eval = 1000, and eval_freq = 1000.

avg_success
--
0.006
0.005
0.011
0.011
0.011
0.015
0.005
0
0.051
0.051
0.002
0.007
0.005
0
0
0
0
0
0
0
0
0
0.001
0.001
0.001
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0

And also, here is my spec file.
```
{
  "rule_dqn": {
    "agent": [{
      "name": "DialogAgent",
      "dst": {
        "name": "RuleDST"
      },
      "state_encoder": {
        "name": "MultiWozStateEncoder"
      },
      "action_decoder": {
        "name": "MultiWozVocabActionDecoder"
      },
      "algorithm": {
        "name": "DQN",
        "action_pdtype": "Argmax",
        "action_policy": "epsilon_greedy",
        "explore_var_spec": {
          "name": "linear_decay",
          "start_val": 0.1,
          "end_val": 0.01,
          "start_step": 0,
          "end_step": 500000
        },
        "gamma": 0.9,
        "training_batch_iter": 1,
        "training_iter": 500,
        "training_frequency": 20,
        "training_start_step": 32
      },
      "memory": {
        "name": "Replay",
        "batch_size": 16,
        "max_size": 50000,
        "use_cer": false
      },
      "net": {
        "type": "MLPNet",
        "hid_layers": [100],
        "hid_layers_activation": "relu",
        "clip_grad_val": null,
        "loss_spec": {
          "name": "MSELoss"
        },
        "optim_spec": {
          "name": "Adam",
          "lr": 0.001
        },
        "lr_scheduler_spec": {
          "name": "StepLR",
          "step_size": 1000,
          "gamma": 0.999
        },
        "update_type": "replace",
        "update_frequency": 300,
        "polyak_coef": 0,
        "gpu": true
      }
    }],
    "env": [{
      "name": "multiwoz",
      "action_dim": 300,
      "observation_dim": 392,
      "max_t": 40,
      "max_frame": 500000,
      "user_policy": {
        "name": "UserPolicyAgendaMultiWoz"
      },
      "sys_policy": {
        "name": "RuleBasedMultiwozBot"
      }
    }],
    "meta": {
      "distributed": false,
      "num_eval": 1000,
      "eval_frequency": 1000,
      "max_tick_unit": "total_t",
      "max_trial": 1,
      "max_session": 1,
      "resources": {
        "num_cpus": 1,
        "num_gpus": 1
      }
    }
  }
}
```
